services/data_service.py

```diff
@@ -92,6 +137,32 @@ def get_repos(self) -> List[Dict[str, str]]:
         repos = self._list(self.REPO_COLLECTION)
         repos_dicts = [{**repo.to_dict(), 'id': repo.id} for repo in repos]
         return repos_dicts
+    
+    def get_user_repo(self, user_id, repo_id) -> Dict[str, Any]:
+        user_repo_query = self._query(self.REPO_COLLECTION, [
+            FirestoreQuery(field_path="owner", op_string=FirestoreQuery.OP_STRING_EQUALS, value=user_id), # query for owner
+            FirestoreQuery(field_path="id", op_string=FirestoreQuery.OP_STRING_EQUALS, value=repo_id)  # query for repo_id
+        ])
+
+        if (len(user_repo_query) > 1):
+            raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                                detail=f"More than one repo found in query.")
+        
+        if (len(user_repo_query) < 1):
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                                detail=f"No repo found with id {repo_id} and owner {user_id}.")
+        
+        repo = user_repo_query[0]
+
+        return {**repo.to_dict(), 'id': repo.id, 'owner': repo.get('owner')}
```

Comment:
**nit**: do we need to manually add `'owner': repo.get('owner')` if we are already doing `**repo.to_dict()`